Riot/Modules/Spaces/SpaceRoomList/SpaceChildRoomDetail/SpaceChildRoomDetailViewModel.swift

```diff
@@ -89,7 +89,23 @@ final class SpaceChildRoomDetailViewModel: SpaceChildRoomDetailViewModelType {
     
     private func joinRoom() {
         self.update(viewState: .loading)
-        self.session.joinRoom(self.childInfo.childRoomId) { [weak self] (response) in
+        if let canonicalAlias = self.childInfo.canonicalAlias {
+            self.session.matrixRestClient.resolveRoomAlias(canonicalAlias) { [weak self] (response) in
+                guard let self = self else { return }
+                switch response {
+                case .success(let resolution):
+                    self.joinRoom(withId: resolution.roomId, via: resolution.servers)
+                case .failure:
+                    self.joinRoom(withId: self.childInfo.childRoomId, via: nil)
```

Comment:
There are two spots here where we pass `nil` for servers, which will not allow a room to be joined by ID if it is on another HS. I assume this will still present some alert error to the user when we call `joinRoom`, but maybe we could log this issue here explicitely?